Statistics/Distributions/Empirical.cs

```diff
@@ -10,22 +10,22 @@ namespace Statistics.Distributions
 {
     class Empirical : IDistribution
     {
-        #region PrivateFields 
+        #region Fields 
         private bool _ProbabilitiesWereAcceptedAsExceedance;
+        internal IRange<double> _ProbabilityRange;
         #endregion
         #region EmpiricalProperties
         /// <summary>
         /// Cumulative probabilities are non-exceedance probabilities 
         /// </summary>
-
         public double[] CumulativeProbabilities;
         public double[] ObservationValues;
         
         #endregion
 
         #region IDistributionProperties
         public IDistributionEnum Type => IDistributionEnum.Empirical;
-
+      
         public double Mean { get 
             {
                 return ComputeMean();
```

Comment:
properties should generally be backed by a field, not computed on the fly. no action needed, just we need to get into the habit of storing information that is accessed in properties rather than computing it for every request.

Reply:
Could you suggest (or point to an example of) a better way to do this?

Originally ComputeMean() was in the constructor, and you suggested that I put this in the property.

Reply:
It looks like I later put this logic in BuildFromProperties()

Reply:
yeah, we want to limit method calls in properties and constructors. again, this really comes from the required methods on idistribution. getting a mean from an emperical is really cool, but not it's primary purpose

it shouldnt be a property, it should be a function, but we can't do that because idistribution requires it to be a property. property should be a reflection of current state (minimal or no computation)
constructors should take arguments to set initial state.
functions should provide matriculated outputs based on state
voids should change state across multiple properties based on arguments injected.